ssl/record/ssl3_record.c

```diff
@@ -683,6 +683,19 @@ int ssl3_get_record(SSL *s)
             }
         }
 
+        /*
+         * A TLS endpoint that receives a record larger than its advertised
+         * limit MUST generate a fatal "record_overflow" alert.
+         * If this extension is not negotiated, endpoints can send
+         * records of any size permitted by the protocol or other negotiated
+         * extensions.
+         */
+        if (s->ext.input_record_size_limit != 0
+                && thisrr->length > s->ext.input_record_size_limit) {
+            SSLfatal(s, SSL_AD_RECORD_OVERFLOW, SSL_R_DATA_LENGTH_TOO_LONG);
```

Comment:
The combination SSL_AD_RECORD_OVERFLOW, SSL_R_DATA_LENGTH_TOO_LONG
already used multiple times, it is admittedly sometimes difficult to see where the RECORD_OVERFLOW
alarm is generated in the server or client, but that alarm code is fixed by the RFCs.
But the diagnostics are easy to decipher, because the come with line numbers.

Reply:
In this case, "Data length longer than Record size limit"  could be more precise.